File: server/php/src/Application/Domain/Payments/PaymentDetails/CustomerBankAccount.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class CustomerBankAccount
{
    public ?string $accountHolderName = null;
    public ?string $bic = null;
    public ?string $iban = null;

    public function toArray(): array
    {
        return [
            'accountHolderName' => $this->accountHolderName,
            'bic' => $this->bic,
            'iban' => $this->iban,
        ];
    }
}

File: server/ruby/app/business/domain/payments/payment_details/customer_bank_account.rb
module Business
  module Domain
    module Payments
      module PaymentDetails
        class CustomerBankAccount
          attr_accessor :account_holder_name, :bic, :iban

          def initialize(account_holder_name: nil, bic: nil, iban: nil)
            @account_holder_name = account_holder_name
            @bic = bic
            @iban = iban
          end

          def to_h
            {
              'accountHolderName' => @account_holder_name,
              'bic' => @bic,
              'iban' => @iban
            }
          end
        end
      end
    end
  end
end
